app/main.py

```diff
@@ -1,3 +1,25 @@
 def cache(func):
-    # Write your code here
-    pass
+    cache_ = {}
+
+    def wrapper(*args):
+        if args in cache_:
+            print("Getting from cache")
+        else:
+            cache_[args] = func(*args)
+            print("Calculating new result")
+        return cache_[args]
+    return wrapper
+
+
+@cache
+def long_time_func(a, b, c):
+    return (a ** b ** c) % (a * c)
+
+
+@cache
+def long_time_func_2(n_tuple, power):
+    return [number ** power for number in n_tuple]
+
+
+long_time_func(1, 2, 3)
+long_time_func(2, 2, 3)
```

Comment:
It's bad practice to leave some test calls of functions in your modules. These lines execute every time you import the module. 
You should either remove these calls or wrap them using `if __name__ == "__main__"` (the sense of this condition is discussed in the `Modules and imports` topic)